runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -474,13 +474,18 @@ function LanguageTree:_async_parse(range, on_parse)
     return
   end
 
-  local buf = vim.b[self._source]
+  local source = self._source
+  local buf = vim.b[source]
   local ct = buf.changedtick
   local total_parse_time = 0
   local redrawtime = vim.o.redrawtime
   local timeout = not vim.g._ts_force_sync_parsing and default_parse_timeout_ms or nil
 
   local function step()
+    if type(source) == 'number' and not vim.api.nvim_buf_is_valid(source) then
```

Comment:
Isn't source always a number?

Reply:
This is to suppress type mismatch diagnosis.

Reply:
Then annotations are the right fix, not "functional" changes.

Reply:
fixed. Test failures is not related this change.

Reply:
Oh sorry, I forgot source could be a string. The code you had before was correct.

Reply:
Oh. In the first place, master branch does not seem to support string source.
This is another issue.

```lua
local parser = assert(vim.treesitter.get_string_parser("", "lua"))
parser:parse(nil, function(...)
  vim.print({ ... })
end)
```
```
E5113: Error while calling lua chunk: ...m/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:478: attempt to index local 'buf' (a nil value)
```
https://github.com/neovim/neovim/blob/851137f67905f6038e51b5b7d1490fbedea4faaa/runtime/lua/vim/treesitter/languagetree.lua#L477-L478

Reply:
Yeah that's a bug.

Reply:
I see.
I've reverted change for now.